### ISSUE TITLE
Refactor NotificationInfo

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#772]
We started refactoring the `NotificationInfo` class in the [`refactor/notification_info` branch.](https://rmc-github-server/common/rafcon/tree/refactor/notification_overview).

The class converts nested notification info objects (of type `dict` or `NTInfo`) into lists. The inner most notification then resides at the end of the list, where it is often accessed with `-1`.

Most of the time, only the inner most notification is of interest, which is why it should be made accessible is a more readably way, e.g. `notification_info.originating_method_name` or just `notification_info.method_name`.

Further helper methods should make the information better accessible:
* `operation_started`/`is_before` (for the "before" case)
* `operation_finished`/`is_after` (for the "after" case)
* `operation_canceled` (if the result is an exception)
* `affects_core_element` (compares the passes element with the "instance")

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2019-07-31 13:57:05+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Here is some information I collected regarding the format of the `info` parameter (only `after` case) when observing different properties of different models and calling `add_state`:

## Observing StateMachineManagerModel.state_machine_manager

### state_machine_manager.add_state_machine()

```
{'after': True,
 'args': (<rafcon.core.state_machine_manager.StateMachineManager object at 0x7fe10a4612e8>,
          <rafcon.core.state_machine.StateMachine object at 0x7fe1030dff98>),
 'instance': <rafcon.core.state_machine_manager.StateMachineManager object at 0x7fe10a4612e8>,
 'kwargs': {},
 'method_name': 'add_state_machine',
 'model': <rafcon.gui.models.state_machine_manager.StateMachineManagerModel object at 0x7fe12b184438>,
 'prop_name': 'state_machine_manager',
 'result': 4}
```

## Observing StateMachineModel.state_machine

### root_state.add_state()

```
{'after': True,
 'args': (<rafcon.core.state_machine.StateMachine object at 0x7f73143d11d0>,),
 'instance': <rafcon.core.state_machine.StateMachine object at 0x7f73143d11d0>,
 'kwargs': {'after': True,
            'args': (<rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f73143abef0>,
                     <rafcon.core.states.execution_state.ExecutionState object at 0x7f73072b2cf8>),
            'instance': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f73143abef0>,
            'kwargs': {},
            'method_name': 'add_state',
            'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f733b534eb8>,
            'prop_name': 'state',
            'result': 'RGQTWA'},
 'method_name': 'root_state_change',
 'model': <rafcon.gui.models.state_machine.StateMachineModel object at 0x7f73143d1c18>,
 'prop_name': 'state_machine',
 'result': None}
```

### child_state.add_state()

```
{'after': True,
 'args': (<rafcon.core.state_machine.StateMachine object at 0x7f73143d11d0>,),
 'instance': <rafcon.core.state_machine.StateMachine object at 0x7f73143d11d0>,
 'kwargs': {'after': True,
            'args': (<rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f73143abef0>,),
            'instance': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f73143abef0>,
            'kwargs': {'after': True,
                       'args': (<rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f7305a39eb8>,
                                <rafcon.core.states.execution_state.ExecutionState object at 0x7f7305d0de80>),
                       'instance': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f7305a39eb8>,
                       'kwargs': {},
                       'method_name': 'add_state',
                       'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7305a38940>,
                       'prop_name': 'state',
                       'result': 'JMEHWZ'},
            'method_name': 'state_change',
            'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f733b534eb8>,
            'prop_name': 'states',
            'result': None},
 'method_name': 'root_state_change',
 'model': <rafcon.gui.models.state_machine.StateMachineModel object at 0x7f73143d1c18>,
 'prop_name': 'state_machine',
 'result': None}
```

### child_state.outcome.name()

```
{'after': True,
 'args': (<rafcon.core.state_machine.StateMachine object at 0x7f72622d5cf8>,),
 'instance': <rafcon.core.state_machine.StateMachine object at 0x7f72622d5cf8>,
 'kwargs': {'after': True,
            'args': (<rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f72622d2fd0>,),
            'instance': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f72622d2fd0>,
            'kwargs': {'after': True,
                       'args': (<rafcon.core.state_elements.logical_port.Outcome object at 0x7f7261531ac8>,
                                'success2'),
                       'instance': <rafcon.core.state_elements.logical_port.Outcome object at 0x7f7261531ac8>,
                       'kwargs': {},
                       'method_name': 'name',
                       'model': <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f72623077f0>,
                       'prop_name': 'outcome',
                       'result': None},
            'method_name': 'state_change',
            'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f72622d5400>,
            'prop_name': 'states',
            'result': None},
 'method_name': 'root_state_change',
 'model': <rafcon.gui.models.state_machine.StateMachineModel object at 0x7f728e18ffd0>,
 'prop_name': 'state_machine',
 'result': None}
```


## Observing StateModel.state

### state.add_state()

```
{'after': True,
 'args': (<rafcon.core.states.hierarchy_state.HierarchyState object at 0x7fa75071d828>,
          <rafcon.core.states.execution_state.ExecutionState object at 0x7fa7502478d0>),
 'instance': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7fa75071d828>,
 'kwargs': {},
 'method_name': 'add_state',
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7fa751f61a20>,
 'prop_name': 'state',
 'result': 'CRNMQD'}
```

### state.outcome.name()

```
{'after': True,
 'args': (<rafcon.core.state_elements.logical_port.Outcome object at 0x7f3c65eef940>,
          'success2'),
 'instance': <rafcon.core.state_elements.logical_port.Outcome object at 0x7f3c65eef940>,
 'kwargs': {},
 'method_name': 'name',
 'model': <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f3c65e835c0>,
 'prop_name': 'outcome',
 'result': None}
```

## Observing StateModel.states

### state.add_state()

```
 {'after': True,
 'args': ('HLHCCH',
          <rafcon.gui.models.state.StateModel object at 0x7f468c0c5e10>),
 'instance': {'BHDEMV': <rafcon.gui.models.state.StateModel object at 0x7f468c226978>,
              'HLHCCH': <rafcon.gui.models.state.StateModel object at 0x7f468c0c5e10>,
              'LMLIMO': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f468c263c50>,
              'PVDEBU': <rafcon.gui.models.state.StateModel object at 0x7f468c22d7b8>,
              'SESNDK': <rafcon.gui.models.state.StateModel object at 0x7f468c230908>,
              'VPKIAE': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f468c243550>},
 'kwargs': {},
 'method_name': '__setitem__',
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f468c27e908>,
 'prop_name': 'states',
 'result': None}
```

### child_state.add_state()

```
 {'after': True,
 'args': ('GSJOIU',
          <rafcon.gui.models.state.StateModel object at 0x7f468c0d5fd0>),
 'instance': {'GSJOIU': <rafcon.gui.models.state.StateModel object at 0x7f468c0d5fd0>,
              'ZCHSQQ': <rafcon.gui.models.state.StateModel object at 0x7f468c1f3c18>},
 'kwargs': {},
 'method_name': '__setitem__',
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f468c263c50>,
 'prop_name': 'states',
 'result': None}

{'after': True,
 'args': (<rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f468c27ec18>,),
 'instance': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f468c27ec18>,
 'kwargs': {'after': True,
            'args': (<rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f468c28cef0>,
                     <rafcon.core.states.execution_state.ExecutionState object at 0x7f468c0679b0>),
            'instance': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f468c28cef0>,
            'kwargs': {},
            'method_name': 'add_state',
            'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f468c263c50>,
            'prop_name': 'state',
            'result': 'GSJOIU'},
 'method_name': 'state_change',
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f468c27e908>,
 'prop_name': 'states',
 'result': None}
```

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2019-08-02 14:20:18+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Now information regarding the `meta_signal`:

## Observing StateMachineModel.meta_signal

### Changing size of root state
```
{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='sm_notification_size',
         affects_children=True,
         notification=Notification(
                        model=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8f9c923518>,
                        prop_name='meta_signal',
                        info={'arg': MetaSignalMsg(
                                       origin='graphical_editor_gaphas',
                                       change='size',
                                       affects_children=True,
                                       notification=None),
                               'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8f9c923518>,
                               'prop_name': 'meta_signal',
                               'signal': True})),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8f9c923518>,
 'prop_name': 'meta_signal',
 'signal': True}

{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='size',
         affects_children=True,
         notification=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8f9c923518>,
 'prop_name': 'meta_signal',
 'signal': True}
```

### Changing size of child state

```
{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='sm_notification_size',
         affects_children=True,
         notification=Notification(
                        model=<rafcon.gui.models.state.StateModel object at 0x7f8f9c068f28>,
                        prop_name='meta_signal',
                        info={'arg': MetaSignalMsg(
                                       origin='graphical_editor_gaphas',
                                       change='size',
                                       affects_children=True,
                                       notification=None),
                               'instance': <rafcon.gui.models.state.StateModel object at 0x7f8f9c068f28>,
                               'method_name': 'change size',
                               'model': <rafcon.gui.models.state.StateModel object at 0x7f8f9c068f28>,
                               'prop_name': 'meta_signal',
                               'signal': True})),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8f9c923518>,
 'prop_name': 'meta_signal',
 'signal': True}
```

## Observing StateMachineModel.state_meta_signal

### Changing size of root state
```
{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='size',
         affects_children=True,
         notification=Notification(
                        model=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6ae245a080>,
                        prop_name='meta_signal',
                        info={'arg': MetaSignalMsg(
                                       origin='graphical_editor_gaphas',
                                       change='size',
                                       affects_children=True,
                                       notification=None),
                               'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6ae245a080>,
                               'prop_name': 'meta_signal',
                               'signal': True})),
 'model': <rafcon.gui.models.state_machine.StateMachineModel object at 0x7f6ae24407f0>,
 'prop_name': 'state_meta_signal',
 'signal': True}
```

### Changing size of child state

```
{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='size',
         affects_children=True,
         notification=Notification(
                        model=<rafcon.gui.models.state.StateModel object at 0x7f6ae33caf98>,
                        prop_name='meta_signal',
                        info={'arg': MetaSignalMsg(
                                       origin='graphical_editor_gaphas',
                                       change='size',
                                       affects_children=True,
                                       notification=None),
                               'instance': <rafcon.gui.models.state.StateModel object at 0x7f6ae33caf98>,
                               'method_name': 'change size',
                               'model': <rafcon.gui.models.state.StateModel object at 0x7f6ae33caf98>,
                               'prop_name': 'meta_signal',
                               'signal': True})),
 'model': <rafcon.gui.models.state_machine.StateMachineModel object at 0x7f6ae24407f0>,
 'prop_name': 'state_meta_signal',
 'signal': True}
```

## Observing StateModel.meta_signal

### changing size of observed root state:

```
{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='size',
         affects_children=True,
         notification=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7bf985bbe0>,
 'prop_name': 'meta_signal',
 'signal': True}

{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='sm_notification_size',
         affects_children=True,
         notification=Notification(
                        model=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7bf985bbe0>,
                        prop_name='meta_signal',
                        info={'arg': MetaSignalMsg(
                                       origin='graphical_editor_gaphas',
                                       change='size',
                                       affects_children=True,
                                       notification=None),
                               'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7bf985bbe0>,
                               'prop_name': 'meta_signal',
                               'signal': True})),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7bf985bbe0>,
 'prop_name': 'meta_signal',
 'signal': True}
```

### changing size of observed child state:

```
{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='size',
         affects_children=True,
         notification=None),
 'model': <rafcon.gui.models.state.StateModel object at 0x7f7bf90184e0>,
 'prop_name': 'meta_signal',
 'signal': True}

{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='size',
         affects_children=True,
         notification=None),
 'model': <rafcon.gui.models.state.StateModel object at 0x7f7bf90184e0>,
 'prop_name': 'meta_signal',
 'signal': True}
```
somehow twice??

### changing size of child state, observing root state:

```
{'arg': MetaSignalMsg(
         origin='graphical_editor_gaphas',
         change='sm_notification_size',
         affects_children=True,
         notification=Notification(
                        model=<rafcon.gui.models.state.StateModel object at 0x7f7bf90184e0>,
                        prop_name='meta_signal',
                        info={'arg': MetaSignalMsg(
                                       origin='graphical_editor_gaphas',
                                       change='size',
                                       affects_children=True,
                                       notification=None),
                               'model': <rafcon.gui.models.state.StateModel object at 0x7f7bf90184e0>,
                               'prop_name': 'meta_signal',
                               'signal': True})),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7bf985bbe0>,
 'prop_name': 'meta_signal',
 'signal': True}
```

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2019-08-02 14:40:56+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Information regarding `action_signal`s

## Observing StateMachineModel.action_signal

### performing type change of root state
```
{'arg': ActionSignalMsg(
         action='sm_notification_change_root_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.state_machine.StateMachineModel object at 0x7f7ca412dc18>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7ccb0b2eb8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7ca4113ba8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7ca41150b8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7ca4115860>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f7ca4113dd8>],
         after=False,
         kwargs={'target_class': <class 'rafcon.core.states.execution_state.ExecutionState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7ccb0b2eb8>,
 'prop_name': 'action_signal',
 'signal': True}

{'arg': ActionSignalMsg(
         action='change_root_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.state_machine.StateMachineModel object at 0x7f7ca412dc18>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7ccb0b2eb8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7ca4113ba8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7ca41150b8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7ca4115860>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f7ca4113dd8>],
         after=False,
         kwargs={'target_class': <class 'rafcon.core.states.execution_state.ExecutionState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7ccb0b2eb8>,
 'prop_name': 'action_signal',
 'signal': True}
```

### performing type change of child state

```
{'arg': ActionSignalMsg(
         action='sm_notification_change_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7c9d168a58>,
         affected_models=[<rafcon.gui.models.state.StateModel object at 0x7f7c9e7307b8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7c9d1eea20>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7c9cb0e4e0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f7c9cb0e710>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f7c9d1eea90>],
         after=False,
         kwargs={'state': <rafcon.core.states.execution_state.ExecutionState object at 0x7f7c9d20b048>,
                  'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f7c9d168a58>,
 'prop_name': 'action_signal',
 'signal': True}
```

## Observing StateMachineModel.state_action_signal

### performing type change of root state
```
{'arg': ActionSignalMsg(
         action='change_root_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.state_machine.StateMachineModel object at 0x7f8ee4b78278>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8f0bd2deb8>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f8ee4b5cc88>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f8ee48fe860>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f8ee48feba8>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f8ee4b5cd30>,
                           <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8ee48fefd0>],
         after=False,
         kwargs={'target_class': <class 'rafcon.core.states.preemptive_concurrency_state.PreemptiveConcurrencyState'>},
         result=None),
 'model': <rafcon.gui.models.state_machine.StateMachineModel object at 0x7f8ee4b78278>,
 'prop_name': 'state_action_signal',
 'signal': True}
```

### performing type change of child state

```
{'arg': ActionSignalMsg(
         action='change_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8ed6d9bf28>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f8ee48fefd0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f8ee4b5c630>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f8ee490f668>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f8ee490f9b0>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f8ee4909b38>],
         after=False,
         kwargs={'state': <rafcon.core.states.hierarchy_state.HierarchyState object at 0x7f8ee4b5c2b0>,
                  'target_class': <class 'rafcon.core.states.preemptive_concurrency_state.PreemptiveConcurrencyState'>},
         result=None),
 'model': <rafcon.gui.models.state_machine.StateMachineModel object at 0x7f8ee4b78278>,
 'prop_name': 'state_action_signal',
 'signal': True}
```

## Observing StateModel.action_signal

### observing root state, performing type change of root state

```
{'arg': ActionSignalMsg(
         action='change_root_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.state_machine.StateMachineModel object at 0x7f658d00ce48>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f65628ca2b0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65628acd68>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65626517f0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f6562651b38>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f65628acda0>,
                           <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>],
         after=False,
         kwargs={'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f65628ca2b0>,
 'prop_name': 'action_signal',
 'signal': True}

{'arg': ActionSignalMsg(
         action='parent_notification_change_root_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.state_machine.StateMachineModel object at 0x7f658d00ce48>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f65628ca2b0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65628acd68>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65626517f0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f6562651b38>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f65628acda0>,
                           <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>],
         after=False,
         kwargs={'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
 'prop_name': 'action_signal',
 'signal': True}

{'arg': ActionSignalMsg(
         action='sm_notification_change_root_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.state_machine.StateMachineModel object at 0x7f658d00ce48>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f65628ca2b0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65628acd68>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65626517f0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f6562651b38>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f65628acda0>,
                           <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>],
         after=False,
         kwargs={'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f65628ca2b0>,
 'prop_name': 'action_signal',
 'signal': True}
```

### observing root state, performing type change of child state

```
{'arg': ActionSignalMsg(
         action='sm_notification_change_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6560ad5cc0>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f656265ca90>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65626624e0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f6562662908>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f656265ceb8>],
         after=False,
         kwargs={'state': <rafcon.core.states.preemptive_concurrency_state.PreemptiveConcurrencyState object at 0x7f65628ac320>,
                  'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6560ad5cc0>,
 'prop_name': 'action_signal',
 'signal': True}
```

### observing child state, performing type change of child state

```
{'arg': ActionSignalMsg(
         action='change_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6560ad5cc0>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f656265ca90>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65626624e0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f6562662908>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f656265ceb8>],
         after=False,
         kwargs={'state': <rafcon.core.states.preemptive_concurrency_state.PreemptiveConcurrencyState object at 0x7f65628ac320>,
                  'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
 'prop_name': 'action_signal',
 'signal': True}

{'arg': ActionSignalMsg(
         action='change_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6560ad5cc0>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f656265ca90>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65626624e0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f6562662908>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f656265ceb8>],
         after=False,
         kwargs={'state': <rafcon.core.states.preemptive_concurrency_state.PreemptiveConcurrencyState object at 0x7f65628ac320>,
                  'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
 'prop_name': 'action_signal',
 'signal': True}

{'arg': ActionSignalMsg(
         action='parent_notification_change_state_type',
         origin='model',
         action_parent_m=<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6560ad5cc0>,
         affected_models=[<rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f656265ca90>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f65626624e0>,
                           <rafcon.gui.models.logical_port.OutcomeModel object at 0x7f6562662908>,
                           <rafcon.gui.models.logical_port.IncomeModel object at 0x7f656265ceb8>],
         after=False,
         kwargs={'state': <rafcon.core.states.preemptive_concurrency_state.PreemptiveConcurrencyState object at 0x7f65628ac320>,
                  'target_class': <class 'rafcon.core.states.hierarchy_state.HierarchyState'>},
         result=None),
 'model': <rafcon.gui.models.container_state.ContainerStateModel object at 0x7f6562651128>,
 'prop_name': 'action_signal',
 'signal': True}
```

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2019-08-05 09:07:58+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Fixed in #784 and #793

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2020-01-27 11:51:52+00:00 (moved from RMC internal repository)
